narwhals/_expression.py

```diff
@@ -28,9 +32,13 @@ def extract_native(expr: Expr, other: Any) -> Any:
 
 
 class Expr:
-    def __init__(self, call: Callable[[Any], Any]) -> None:
+    def __init__(self, call: Callable[[Any], Any], api_version: API_VERSION) -> None:
         # callable from namespace to expr
         self._call = call
+        self._api_version = api_version
```

Comment:
I see that when `Expr` is instantiated below, we do many `api_version or DEFAULT_API_VERSION`.
Would it make sense to move the logic here?
like
```python
class Expr
    def __init__(self, call: Callable[[Any], Any], api_version: API_VERSION|None = None) -> None:
        ...
        self._api_version = api_version or DEFAULT_API_VERSION
```

Reply:
yeah probably